privilege/privileges/privileges.go

```diff
@@ -323,7 +381,21 @@ func (p *UserPrivileges) ConnectionVerification(user *auth.UserIdentity, authUse
 		return ErrAccessDenied.FastGenByArgs(user.Username, user.Hostname, hasPassword)
 	}
 
-	if len(pwd) > 0 && len(authentication) > 0 {
+	if record.AuthPlugin == mysql.AuthTiDBAuthToken {
+		tokenString := string(authentication[:len(authentication)-1])
```

Comment:
Are you sure `len(authentication) > 0`?

Reply:
Yes. This `authentication` comes from  `func (cc *clientConn) authSwitchRequest`, which return a [Protocol::AuthSwitchResponse](https://dev.mysql.com/doc/dev/mysql-server/latest/page_protocol_connection_phase_packets_protocol_auth_switch_response.html) that ends with digital '0'.

But I think it is indeed better to add a length check here.